### PR TITLE
fix: naming for model registry CLI commands and entities [DET-3767]

### PR DESCRIPTION
**Pull request by sidneyw**
## Description
Changes `version` to `model_version` in the REST API
Changes underscores to hyphens in the CLI

### CODE REVIEW

File: common/determined_common/experimental/model.py [hunk 1]
```diff
@@ -97,16 +97,19 @@ def get_versions(self, order_by: ModelOrderBy = ModelOrderBy.DESC) -> List[Check
         )
         data = resp.json()
 
+        for v in data["modelVersions"]:
```
Comment (neilconway) [inline common/determined_common/experimental/model.py:100]: Is this debug code?

File: master/internal/api_model.go [hunk 2]
```diff
@@ -123,12 +124,15 @@ func (a *apiServer) GetModelVersions(
...
-	if err := a.m.db.QueryProto("get_model_versions", &resp.Versions, req.ModelName); err != nil {
+	if err := a.m.db.QueryProto("get_model_versions", &resp.ModelVersions, req.ModelName); err != nil {
 		return nil, err
 	}
 
-	a.sort(resp.Versions, req.OrderBy, req.SortBy, apiv1.GetModelVersionsRequest_SORT_BY_VERSION)
-	return resp, a.paginate(&resp.Pagination, &resp.Versions, req.Offset, req.Limit)
+	a.sort(resp.ModelVersions, req.OrderBy, req.SortBy, apiv1.GetModelVersionsRequest_SORT_BY_VERSION)
+	for _, v := range resp.ModelVersions {
+		fmt.Printf("v: %v, uuid: %v\n", v.Version, v.Checkpoint.Uuid)
```
Comment (neilconway) [inline master/internal/api_model.go:133]: Debug code?

File: cli/determined_cli/model.py [hunk 3]
```diff
@@ -169,7 +170,7 @@ def register_version(args: Namespace) -> None:
                 is_default=True,
             ),
             Cmd(
-                "register_version",
+                "register-version",
```
Comment (aaron276h) [inline cli/determined_cli/model.py:173]: question: should we make this `register-model-verion`?

Comment (neilconway) [reply cli/determined_cli/model.py:173]: The CLI command already has `model` in it (`det model register-version`), so personally I think `register-model-version` would be redundant.

**Pull request merged**